### PR TITLE
refactor(ui/lineage): Replace FetchedEntities Object with Map

### PR DESCRIPTION
**Pull request by asikowitz**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts [hunk 1]
```diff
@@ -198,11 +198,7 @@ export default function extendAsyncEntities(
...
     extendColumnLineage(lineageVizConfig, fineGrainedMap, fineGrainedMapForSiblings, fetchedEntities);
 
-    return {
-        ...fetchedEntities,
-        [entityAndType.entity.urn]: {
-            ...lineageVizConfig,
-            fullyFetched,
-        },
-    };
+    const newFetchedEntities = new Map(fetchedEntities);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts:201]: is this line safe?

File: datahub-web-react/src/app/lineage/utils/columnLineageUtils.ts [hunk 2]
```diff
@@ -71,10 +71,10 @@ export function convertInputFieldsToSchemaFields(inputFields?: InputFields) {
  */
 export function getPopulatedColumnsByUrn(
     columnsByUrn: Record<string, SchemaField[]>,
-    fetchedEntities: { [x: string]: FetchedEntity },
+    fetchedEntities: Map<string, FetchedEntity>,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/utils/columnLineageUtils.ts:74]: nice - okay change below i think would be safe if this is what is getting passed

**Review by jjoyce0510 — APPROVED** [review]
If you've tested everything locally with a non-trivial lineage graph

File: datahub-web-react/src/app/lineage/LineageExplorer.tsx [hunk 3]
```diff
@@ -77,12 +77,12 @@ export default function LineageExplorer({ urn, type }: Props) {
 
     const [isDrawerVisible, setIsDrawVisible] = useState(false);
     const [selectedEntity, setSelectedEntity] = useState<EntitySelectParams | undefined>(undefined);
-    const [asyncEntities, setAsyncEntities] = useState<FetchedEntities>({});
+    const [asyncEntities, setAsyncEntities] = useState<FetchedEntities>(new Map());
```
[the remark this answers is not in the document]
Comment (asikowitz) [reply datahub-web-react/src/app/lineage/LineageExplorer.tsx:80]: These are equivalent: `type FetchedEntities = Map<string, FetchedEntity>`

**Pull request merged**